pkg/reconciler/apibinding/apibinding_reconcile_test.go

```diff
@@ -0,0 +1,745 @@
+							Storage:            false,
+							Deprecated:         true,
+							DeprecationWarning: pointer.StringPtr("deprecated!"),
+							Schema: &apiextensionsv1.CustomResourceValidation{
+								OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+									Description: "foo",
+									Type:        "object",
+								},
+							},
+							Subresources: &apiextensionsv1.CustomResourceSubresources{
+								Status: &apiextensionsv1.CustomResourceSubresourceStatus{},
+								Scale: &apiextensionsv1.CustomResourceSubresourceScale{
+									SpecReplicasPath:   ".spec.replicas",
+									StatusReplicasPath: ".status.replicas",
+									LabelSelectorPath:  pointer.StringPtr(".status.selector"),
+								},
+							},
+							AdditionalPrinterColumns: []apiextensionsv1.CustomResourceColumnDefinition{
+								{
+									Name:        "My Column",
+									Type:        "string",
+									Format:      "string",
+									Description: "This is my column",
+									Priority:    1,
+									JSONPath:    ".spec.myColumn",
+								},
+							},
+						},
+						{
+							Name:       "v2",
+							Served:     true,
+							Storage:    true,
+							Deprecated: false,
+							Schema: &apiextensionsv1.CustomResourceValidation{
+								OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+									Description: "foo",
+									Type:        "object",
+								},
+							},
+							Subresources: &apiextensionsv1.CustomResourceSubresources{
+								Status: &apiextensionsv1.CustomResourceSubresourceStatus{},
+								Scale: &apiextensionsv1.CustomResourceSubresourceScale{
+									SpecReplicasPath:   ".spec.replicas",
+									StatusReplicasPath: ".status.replicas",
+									LabelSelectorPath:  pointer.StringPtr(".status.selector"),
+								},
+							},
+							AdditionalPrinterColumns: []apiextensionsv1.CustomResourceColumnDefinition{
+								{
+									Name:        "My Column",
+									Type:        "string",
+									Format:      "string",
+									Description: "This is my column",
+									Priority:    1,
+									JSONPath:    ".spec.myColumn",
+								},
+							},
+						},
+					},
+					Conversion:            nil,
+					PreserveUnknownFields: false,
+				},
+			},
+			wantErr: false,
+		},
+		"error when schema is invalid": {
+			schema: &apisv1alpha1.APIResourceSchema{
+				Spec: apisv1alpha1.APIResourceSchemaSpec{
+					Versions: []apisv1alpha1.APIResourceVersion{
+						{
+							Schema: runtime.RawExtension{
+								Raw: []byte("invalid json"),
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for testName, tc := range tests {
+		t.Run(testName, func(t *testing.T) {
+			got, err := crdFromAPIResourceSchema(tc.schema)
+
+			if tc.wantErr != (err != nil) {
+				t.Fatalf("wantErr: %v, got %v", tc.wantErr, err)
+			}
+			if tc.wantErr {
+				return
+			}
+
+			require.Equal(t, tc.want, got)
+		})
+	}
+}
+
+func TestWorkspaceAPIExportReferenceReconciler(t *testing.T) {
+	tests := map[string]struct {
```

Comment:
(No action required) The complexity of test case definition suggests that the functionality under test would benefit from refactoring for testability. If not in this PR, then in the near future.